pkg/operator/ceph/config/conditions.go

```diff
@@ -117,28 +117,32 @@ func translatePhasetoState(phase cephv1.ConditionType) cephv1.ClusterState {
 	}
 }
 
-// Updating the status of Progressing, Updating or Upgrading to False once cluster is Ready
-func checkConditionFalse(context *clusterd.Context, namespace, name string) {
+// MakeConditionFalse Updates the status of Progressing, Updating or Upgrading to False once cluster is Ready
+func MakeConditionFalse(context *clusterd.Context, namespace, name string) {
 	tempConditionList := []cephv1.ConditionType{cephv1.ConditionUpdating, cephv1.ConditionUpgrading, cephv1.ConditionProgressing}
-	var tempCondition cephv1.ConditionType
+	var tempCondition []cephv1.ConditionType
 	for _, conditionType := range tempConditionList {
 		if conditionMap[conditionType] == v1.ConditionTrue {
-			tempCondition = conditionType
+			tempCondition = append(tempCondition, conditionType)
 		}
 	}
-	reason := ""
-	message := ""
-	if tempCondition == cephv1.ConditionUpdating {
-		reason = "UpdateCompleted"
-		message = "Cluster updating is completed"
-	} else if tempCondition == cephv1.ConditionUpgrading {
-		reason = "UpgradeCompleted"
-		message = "Cluster upgrading is completed"
-	} else {
-		reason = "ProgressingCompleted"
-		message = "Cluster progression is completed"
+	for _, conditionType := range tempCondition {
```

Comment:
Why do we need this separate `for` loop? It seems simpler if you combined the two loops and just set the condition immediately instead of having the temporary slice created.

Reply:
Combined both of them.